smart-contracts/test/Forwarder/forwarder.hardhat.ts

```diff
@@ -0,0 +1,239 @@
+import { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { SignerWithAddress } from '@nomicfoundation/hardhat-ethers/signers'
+import { Forwarder, MockSafe, MockTarget } from '../../typechain-types'
+import { BaseContract } from 'ethers'
+
+async function deployContract<T extends BaseContract>(
+  name: string,
+  args: any[] = []
+): Promise<T> {
+  const Contract = await ethers.getContractFactory(name)
+  const contract = await Contract.deploy(...args)
+  await contract.waitForDeployment()
+  return contract as unknown as T
+}
+
+async function deployMockSafe(owners: string[]): Promise<MockSafe> {
+  const MockSafe = await ethers.getContractFactory('MockSafe')
+  const mockSafe = await MockSafe.deploy(owners)
+  await mockSafe.waitForDeployment()
+  return mockSafe as unknown as MockSafe
+}
```

Comment:
Do we really need this? If we have `deployContract` (which I am not sure we really need), then we certainly don't need `deployMockSafe`...